packages/reakit/src/Sidebar/Sidebar.tsx

```diff
@@ -15,7 +19,7 @@ const Component = props => (
 
 hoistNonReactStatics(Component, Overlay);
 
-const Sidebar = styled(Component)`
+const Sidebar = styled<SidebarProps>(Component)`
```

Comment:
Does it work if you put `<SidebarProps>` at the end?
```
const Sidebar = styled(Component)<SidebarProps>`
```

Or better, wouldn't it infer `SidebarProps` from `Component`, making the cast unnecessary? 🤔

You can use `packages/reakit/test.tsx` to try it.

Reply:
Yes, you're right:

<img width="617" alt="image" src="https://user-images.githubusercontent.com/827338/45584610-8b5ac680-b8df-11e8-8c5c-cf0d9b75ca87.png">

Thanks!